testing.rst

```diff
@@ -11,377 +11,513 @@ using both functional and unit tests.
-wraps the original PHPUnit binary to provide additional features:
+Before creating your first test, install PHPUnit with the following
+command:
 
 .. code-block:: terminal
 
-    $ composer require --dev symfony/phpunit-bridge
+    $ composer require --dev phpunit/phpunit
 
-After the library downloads, try executing PHPUnit by running (the first time
-you run this, it will download PHPUnit itself and make its classes available in
-your app):
+After the library is installed, try running PHPUnit:
 
 .. code-block:: terminal
 
-    $ ./bin/phpunit
+    $ php ./vendor/bin/phpunit
+
+This commands automatically runs your application's tests. Each test is a
+PHP class ending with "Test" (e.g. ``BlogControllerTest``) that lives in
+the ``tests/`` directory of your application.
+
+PHPUnit is configured by the ``phpunit.xml.dist`` file in the root of your
+application.
 
 .. note::
 
-    The ``./bin/phpunit`` command is created by :ref:`Symfony Flex <symfony-flex>`
-    when installing the ``phpunit-bridge`` package. If the command is missing, you
-    can remove the package (``composer remove symfony/phpunit-bridge``) and install
-    it again. Another solution is to remove the project's ``symfony.lock`` file and
-    run ``composer install`` to force the execution of all Symfony Flex recipes.
+    :ref:`Symfony Flex <symfony-flex>` automatically creates
+    ``phpunit.xml.dist`` and ``tests/bootstrap.php``. If these files are
+    missing, you can try running the recipe again using
+    ``composer recipes:install phpunit/phpunit --force -v``.
 
-Each test - whether it's a unit test or a functional test - is a PHP class
-that should live in the ``tests/`` directory of your application. If you follow
-this rule, then you can run all of your application's tests with the same
-command as before.
+Types of Tests
+--------------
 
-PHPUnit is configured by the ``phpunit.xml.dist`` file in the root of your
-Symfony application.
+There are many types of automated tests and precise definitions often
+differ from project to project. In Symfony, the following definitions are
+used. If you have learned something different, that is not necessarily
+wrong, just different from what the Symfony documentation is using.
 
-.. tip::
+`Unit Tests`_
+    These tests ensure that *individual* units of source code (e.g. a single
+    class) behave as intended.
 
-    Use the ``--coverage-*`` command options to generate code coverage reports.
-    Read the PHPUnit manual to learn more about `code coverage analysis`_.
+`Integration Tests`_
+    These tests test a combination of classes and commonly interact with
+    Symfony's service container. These tests do not yet cover the fully
+    working application, those are called *Application tests*.
 
-.. index::
-   single: Tests; Unit tests
+`Application Tests`_
+    Application tests test the behavior of a complete application. They
+    make, sometimes fake, HTTP requests and test that the response is as
```

Comment:
"Faked" requests is super confusing as a new user. Lets put it in quotation marks to show that this is a concept. 

```suggestion
    make HTTP requests (both real and "faked") and test that the response is as
```

Reply:
:+1: also used "simulated" instead of "faked" :)

Reply:
Good. That is a better word.